tests/bfd/test_bfd.py

```diff
@@ -40,61 +41,97 @@ def get_t0_intfs(mg_facts):
-            neighbor_interfaces = neighbor_devs
-        else:
-            vlan_subnet = ipaddress.ip_network(vlan_intf['subnet'])
-            vlan = mg_facts['minigraph_vlans'][mg_facts['minigraph_vlan_interfaces'][1 if ipv6 else 0]['attachto']]
-            vlan_ports = vlan['members']
-            vlan_id = vlan['vlanid']
-            vlan_ptf_ports = [mg_facts['minigraph_ptf_indices'][port] for port in vlan_ports]
-            neighbor_devs = vlan_ptf_ports
-            # backend topology use ethx.x(e.g. eth30.1000) during servers and T0 in ptf
-            # in other topology use ethx(e.g. eth30)
-            if is_backend_topology:
-                neighbor_interfaces = [str(dev) + constants.VLAN_SUB_INTERFACE_SEPARATOR + str(vlan_id) for dev in neighbor_devs]
-            else:
-                neighbor_interfaces = neighbor_devs
-            neighbor_addrs = [str(vlan_subnet[i + 2]) for i in range(len(neighbor_devs))]
-        count = min(count, len(neighbor_devs))
-        indices = random.sample(list(range(len(neighbor_devs))), k=count)
-        return [vlan_addr for _ in indices], prefix_len, [neighbor_addrs[_] for _ in indices], [neighbor_devs[_] for _ in indices], [neighbor_interfaces[_] for _ in indices]
-    elif 't1' in topo_type:
+    if 't1' in topo_type:
         t1_ipv4_pattern = '101.0.0.{}'
         t1_ipv6_pattern = '2000:2000::{:x}'
         t0_intfs = get_t0_intfs(mg_facts)
         ptf_ports = [mg_facts['minigraph_ptf_indices'][port] for port in t0_intfs]
         count = min(count, len(t0_intfs))
         indices = random.sample(list(range(len(t0_intfs))), k=count)
+        port_intfs = [t0_intfs[_] for _ in indices]
+        neighbour_interfaces =[]
+        for intf in port_intfs:
+            pc_member = False
+            for pc in mg_facts['minigraph_portchannels']:
+                if intf in mg_facts['minigraph_portchannels'][pc]['members']:
+                    neighbour_interfaces.append(pc)
+                    pc_member= True
+                    break
+            if not pc_member:
+                neighbour_interfaces.append(intf)
         if ipv6:
-            return [t1_ipv6_pattern.format(idx * 2) for idx in indices], 127, [t1_ipv6_pattern.format(idx * 2 + 1) for idx in indices], [t0_intfs[_] for _ in indices], [ptf_ports[_] for _ in indices]
+            return [t1_ipv6_pattern.format(idx * 2) for idx in indices], 127, [t1_ipv6_pattern.format(idx * 2 + 1) for idx in indices], neighbour_interfaces, [ptf_ports[_] for _ in indices]
         else:
-            return [t1_ipv4_pattern.format(idx * 2) for idx in indices], 31, [t1_ipv4_pattern.format(idx * 2 + 1) for idx in indices], [t0_intfs[_] for _ in indices], [ptf_ports[_] for _ in indices]
-
+            return [t1_ipv4_pattern.format(idx * 2) for idx in indices], 31, [t1_ipv4_pattern.format(idx * 2 + 1) for idx in indices], neighbour_interfaces, [ptf_ports[_] for _ in indices]
```

Comment:
It's better to have one return statement. And the amount of returning variables is too big. I'd suggest creating a dictionary, populating it, and then returning only it.